library/cartridge_set_dist_dirs_to_remove.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/python
+
+import os
+import pkgutil
+
+argument_spec = {
+    'app_name': {'required': False, 'type': 'str'},
+    'app_install_dir': {'required': True, 'type': 'str'},
+    'dists_retention_num': {'required': True, 'type': 'int'},
+}
+
+if pkgutil.find_loader('ansible.module_utils.helpers'):
+    import ansible.module_utils.helpers as helpers
+else:
+    import module_utils.helpers as helpers
+
+
+def is_dist(filename, app_install_dir, app_name):
+    if not os.path.isdir(os.path.join(app_install_dir, filename)):
+        return False
+
+    if not filename.startswith('%s-' % app_name):
+        return False
+
+    return True
+
+
+def get_dist_dirs_to_remove(params):
+    app_name = params['app_name']
+    dists_retention_num = params['dists_retention_num']
+    app_install_dir = params['app_install_dir']
+
+    if dists_retention_num <= 0:
+        return helpers.ModuleRes(failed=True, msg='"dists_retention_num" should be > 0')
+
+    dists = list(filter(
+        lambda filename: is_dist(filename, app_install_dir, app_name),
+        os.listdir(app_install_dir)
+    ))
+
+    dists = sorted(
+        dists,
+        key=lambda filename: os.path.getmtime(os.path.join(app_install_dir, filename)),
```

Comment:
Is it possible that 1.0.0 will be newer than 2.0.0?

Reply:
I don't know, but mtime allows us to be sure always.